pkg/manager/status/status_manager.go

```diff
@@ -173,6 +178,52 @@ func checkValidation(ctx context.Context, fc *flowslatest.FlowCollector) metav1.
 	}
 }
 
+func checkLoki(ctx context.Context, c client.Client, fc *flowslatest.FlowCollector) metav1.Condition {
+	if !helper.UseLoki(&fc.Spec) {
+		return metav1.Condition{
+			Type:    LokiIssue,
+			Reason:  "Unused",
+			Status:  metav1.ConditionUnknown,
+			Message: "Loki is disabled",
+		}
+	}
+	if fc.Spec.Loki.Mode != flowslatest.LokiModeLokiStack {
+		return metav1.Condition{
+			Type:    LokiIssue,
+			Reason:  "Unused",
+			Status:  metav1.ConditionUnknown,
+			Message: "Loki is not configured in LokiStack mode",
+		}
+	}
+	lokiStack := &lokiv1.LokiStack{}
+	nsname := types.NamespacedName{Name: fc.Spec.Loki.LokiStack.Name, Namespace: fc.Spec.Namespace}
+	if len(fc.Spec.Loki.LokiStack.Namespace) > 0 {
+		nsname.Namespace = fc.Spec.Loki.LokiStack.Namespace
+	}
+	err := c.Get(ctx, nsname, lokiStack)
+	if err != nil {
+		if kerr.IsNotFound(err) {
+			return metav1.Condition{
+				Type:    LokiIssue,
+				Reason:  "LokiStackNotFound",
+				Status:  metav1.ConditionTrue,
+				Message: fmt.Sprintf("The configured LokiStack reference could not be found [name: %s, namespace: %s]", nsname.Name, nsname.Namespace),
+			}
+		}
+		return metav1.Condition{
+			Type:    LokiIssue,
+			Reason:  "Error",
+			Status:  metav1.ConditionTrue,
+			Message: fmt.Sprintf("Error while fetching configured LokiStack: %s", err.Error()),
+		}
+	}
+	return metav1.Condition{
+		Type:   LokiIssue,
+		Reason: "NoIssue",
+		Status: metav1.ConditionFalse,
+	}
```

Comment:
Why not going one step further here checking LokiStack status and mode ?

Reply:
sounds good!
=> https://issues.redhat.com/browse/NETOBSERV-2189